tests/compiler/binary.untouched.wat

```diff
@@ -2583,7 +2585,9 @@
    local.set $2
   end
   local.get $5
-  i64.eqz
+  i64.const 0
+  i64.ne
+  i32.eqz
```

Comment:
These ones are a bit unexpected, not sure why these aren't properly optimized.

Reply:
yeah, binaryen have some missing peephole optimizations with mixed series `i32.ne`/`i64.eqz` and etc. But probably @kripken know better what happened

Reply:
`(i32.eqz (i64.ne ..))` should definitely be optimized to `(i64.eq ..)`.

Is this "untouched" file after binaryen optimizations? And which opts?

Reply:
Ah, sorry this unoptimized files

But for optimized it seems we have minor regressions like
```wat
i64.eqz
```
to
```wat
i64.const 0
i64.eq
```

https://github.com/AssemblyScript/assemblyscript/pull/933/commits/cbb83744e9d46a565cab8f477497a80849f984c0#diff-587a89b754ca1898d922a9e2c289bcfcR96

Reply:
That should be optimized by the precompute pass. The only way it can't is if the constants are generated by very late passes in the optimization pipeline. Running another `-O` (or `--converge`) should handle this.

Reply:
Thanks. I guess it will be optimized after #929 PR via extra [`inlining-optimizing` pass](https://github.com/AssemblyScript/assemblyscript/pull/929/files#diff-1d2bb562554675174e389bb738b930cdR710) anyway

Reply:
Oh, yeah intended to refer to [those](https://github.com/AssemblyScript/assemblyscript/commit/cbb83744e9d46a565cab8f477497a80849f984c0#diff-587a89b754ca1898d922a9e2c289bcfcR96)

```wat
i64.const 0
i64.eq
```

in optimized files. Maybe that's replacing something which isn't visited again, or it's happening too late.

Reply:
Taking a quick look at `OptimizeInstructions`, it seems that while there are quite a few `i32.eqz` optimizations, I am struggling to find the same for `i64.eqz`. Might that be the issue here?

```wat
i64.const 0
i64.ne  ; -> i64.eq
i32.eqz ;
```

with a missing case of

```wat
i64.const 0 ; -> i64.eqz
i64.eq      ; 
```